pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -491,27 +491,31 @@ func (c *Controller) updateGateways(serviceList *federationmodel.ServiceListMess
 }
 
 func (c *Controller) getEgressServiceAddrs() ([]model.NetworkGateway, []string) {
-	endpoints, err := common.EndpointsForService(c.kubeClient, c.egressName, c.namespace)
+	endpointSlices, err := common.EndpointSlicesForService(c.kubeClient, c.egressName, c.namespace)
 	if err != nil {
-		c.logger.Errorf("unabled to retrieve endpoints for federation egress gateway %s: %s", c.egressName, err)
+		c.logger.Errorf("failed to retrieve EndpointSlices for federation egress gateway %s: %s", c.egressName, err)
 		return nil, nil
 	}
 	serviceAccountByIP := map[string]string{}
 	if !c.useDirectCalls {
 		serviceAccountByIP, err = common.ServiceAccountsForService(c.kubeClient, c.egressName, c.namespace)
 		if err != nil {
-			c.logger.Warnf("unable to retrieve ServiceAccount information for imported services accessed through %s: %s",
+			c.logger.Warnf("failed to retrieve ServiceAccount information for imported services accessed through %s: %s",
 				c.egressName, err)
 		}
 	}
 	var addrs []model.NetworkGateway
 	var sas []string
-	for _, subset := range endpoints.Subsets {
-		for index, address := range subset.Addresses {
-			if subset.Ports[index].Port == common.DefaultFederationPort {
```

Comment:
There is no need to look for a port under an index, because the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/endpoint-slice-v1/) of EndpoinSlices says:
```
ports specifies the list of network ports exposed by each endpoint in this slice
```
so it means that all addresses in all endpoints from a given EndpointSlice object expose ports specified in the `Ports` array.

Reply:
I will also fix it in the previous versions.